Assets/HoloToolkit/Input/Scripts/Utilities/MicrophoneHelperSample.cs

```diff
@@ -1,159 +1,34 @@
-                // Can also occur if using the virtual machine console tool to access a VM instead of using remote desktop.
-                // You can also get here if you disable all audio recording endpoints
-                if (exception.HResult == NoCaptureDevicesHResult)
-                {
-                    return MicrophoneStatus.MicrophoneNotPresent;
-                }
-                else
-                {
-                    return MicrophoneStatus.MicrophoneUseFailed;
-                }
-            }
-
-            // if microphone is muted, speech recognition will not hear anything.
-            // return appropriate value to indicate that.
-            if(capture.AudioDeviceController.Muted)
-            {
-                return MicrophoneStatus.MicrophoneMuted;
-            }
-
-            return MicrophoneStatus.MicrophoneReady;
-        }
-
-        /// <summary>
-        /// This function provides an easier query when one is only interested to know if the microphone
-        /// can be used or not, without dealing with the specifics of why it may not be working
-        /// </summary>
-        /// <returns>
-        /// Returns true if the microphone is accessible and ready for use.
-        /// </returns>
-        public static async Task<bool> IsMicrophoneReady()
-        {
-            var status = await GetMicrophoneStatus();
-            return status == MicrophoneStatus.MicrophoneReady;
-        }
-#else
-        // if we run in the editor return that microphone is ready by default
-        // It could be useful to test in the editor the behavior where the
-        // microphone is reported ready.
-        public static MicrophoneStatus GetMicrophoneStatus()
-        {
-            return MicrophoneStatus.MicrophoneReady;
-        }
-
-        public static bool IsMicrophoneReady()
-        {
-            var status = GetMicrophoneStatus();
-            return status == MicrophoneStatus.MicrophoneReady;
-        }
-#endif
-    }
-}
-
-public class MicrophoneHelperSample : MonoBehaviour
+namespace HoloToolkit.Unity
 {
-    [Tooltip("Assign key to quickly test the microphone helper code")]
-    public KeyCode checkMicrophoneKey = KeyCode.M;
+    public class MicrophoneHelperSample : MonoBehaviour
+    {
+        [Tooltip("Assign key to quickly test the microphone helper code")]
+        public KeyCode checkMicrophoneKey = KeyCode.M;
 #if ENABLE_WINMD_SUPPORT
-    async void Update ()
+        private async void Update()
 #else
-    void Update()
+        private void Update()
 #endif
-    {
-        if(Input.GetKeyDown(checkMicrophoneKey))
         {
+            if(Input.GetKeyDown(checkMicrophoneKey))
+            {
 #if ENABLE_WINMD_SUPPORT
-            var status = await MixedRealityToolkit.MicrophoneHelper.GetMicrophoneStatus();
+                var status = await MicrophoneHelper.GetMicrophoneStatus();
```

Comment:
Is there a way we can use `MicrophoneHelper.GetMicrophoneStatus()` without having to check for `ENABLE_WINMD_SUPPORT` in our implementation?

Reply:
I could also make the stub method async, but then I'll get a warning that await is not being used anywhere. If you're fine with an extra warning I can change that.

Reply:
When will the warning show? During compile or build time?

Reply:
If compile time, I'd rather not, but if it's at build time, then it's not that big a deal.

Reply:
Actually this is not a good idea. async/await were introduces with C# 5 and require .Net 4.5+. It would be better it remained this way.

Reply:
Just trying to think about ways to make this simpler for devs to use in their own code.